paimon-common/src/main/java/org/apache/paimon/format/FormatWriter.java

```diff
@@ -60,4 +60,27 @@ public interface FormatWriter {
      * @throws IOException Thrown if the finalization fails.
      */
     void finish() throws IOException;
+
+    /**
+     * Returns the number of bytes that were currently written by this writer.
+     *
+     * <p>NOTE: This is an estimated value.
+     *
+     * @return the number of written bytes
+     * @throws IOException Thrown if calculating the length fails.
+     */
+    long length() throws IOException;
```

Comment:
We don't need to introduce this method.

Reply:
Removed and moved implementation into reachTargetSize.